src/components/src/side-panel/layer-panel/color-range-selector.tsx

```diff
@@ -94,143 +103,171 @@ const CONFIG_SETTINGS = {
-          <CustomPalette
-            customPalette={customPalette}
-            showSketcher={showSketcher}
-            onApply={this.props.onSelectColorRange}
-            onToggleSketcher={this._onToggleSketcher}
-            setCustomPalette={this._onSetCustomPalette}
-            onCancel={this._onCustomPaletteCancel}
-          />
+          <>
+            <StyledColorPalette>
+              <ColorPalette colors={customPalette.colors} />
+            </StyledColorPalette>
+            <CustomPalette
+              customPalette={customPalette}
+              showSketcher={showSketcher}
+              onApply={_onApply}
+              setColorPaletteUI={setColorPaletteUI}
+              onCancel={_onCustomPaletteCancel}
+            />
+          </>
         ) : (
           <ColorPaletteGroup
             colorRanges={filteredColorRanges}
-            onSelect={this.props.onSelectColorRange}
-            selected={this.props.selectedColorRange}
+            onSelect={onSelectFromList}
+            selected={selectedColorRange}
             reversed={colorRangeConfig.reversed}
           />
         )}
       </StyledColorRangeSelector>
     );
-  }
+  };
+
+  ColorRangeSelector.defaultProps = {
+    colorRanges: COLOR_RANGES,
+    onSelectColorRange: () => {},
+    setColorPaletteUI: () => {}
+  };
+
+  return ColorRangeSelector;
 }
 
 export const PaletteConfig: React.FC<PaletteConfigProps> = ({
+  prop,
   label,
   value,
   config: {type, options},
-  onChange
-}) => (
-  <StyledPaletteConfig className="color-palette__config" onClick={e => e.stopPropagation()}>
-    <div className="color-palette__config__label">
-      <PanelLabel>
-        <FormattedMessage id={`color.${label}`} />
-      </PanelLabel>
-    </div>
-    {type === 'select' && (
-      <div className="color-palette__config__select">
-        <ItemSelector
-          selectedItems={value}
-          options={options}
-          multiSelect={false}
-          searchable={false}
-          onChange={onChange}
-        />
-      </div>
-    )}
-    {type === 'switch' && (
-      <Switch
-        checked={value as boolean}
-        id={`${label}-toggle`}
-        onChange={() => onChange(!value)}
-        secondary
-      />
-    )}
-  </StyledPaletteConfig>
-);
+  onChange,
+  disabled,
+  reason
+}) => {
+  const updateSelect = useCallback(val => onChange({key: prop, value: val}), [onChange, prop]);
+  const updateSwitch = useCallback(
+    () => onChange({key: prop, value: !value}),
+    [onChange, prop, value]
+  );
 
+  return (
+    <StyledPaletteConfig className="color-palette__config" onClick={e => e.stopPropagation()}>
+      <div className="color-palette__config__label">
+        <PanelLabel>
+          <FormattedMessage id={`color.${label}`} />
+        </PanelLabel>
+      </div>
+      <div
+        className="color-palette__config__select"
+        data-tip
+        data-for={`color-range-config-${prop}`}
+      >
+        {type === 'select' && (
```

Comment:
Nit: a switch statement on a field that can have certain string values seems preferable to ad-hoc if statements.